src/solve.jl

```diff
@@ -75,7 +75,7 @@ function init(prob::AbstractDDEProblem{uType,tType,lType,isinplace}, alg::algTyp
                                                        ode_prob,
                                                        OrdinaryDiffEq.alg_order(alg)))
     end
-    integrator.dt = dt
+    integrator.dt = zero(dt)
```

Comment:
I'm not sure I understand you. Of course, for all other steps (besides the initial step) lines https://github.com/devmotion/DelayDiffEq.jl/blob/8008e8f2d0837c944e7a7093a948a4fb094f0bb5/src/integrator_utils.jl#L239-L242 and https://github.com/devmotion/DelayDiffEq.jl/blob/8008e8f2d0837c944e7a7093a948a4fb094f0bb5/src/integrator_interface.jl#L70-L72 guarantee that `t`, `tprev`, and `dt`of the ODE integrator `integrator.integrator` belong to the same time interval which is important for correct interpolation. So setting `dt` to zero initially (only for `integrator.integrator`, the DDE integrator still uses the supplied or calculated `dt`!) since `tprev = t` initially is not needed for any calculations but ensures this consistent relationship of these three fields throughout the whole algorithm.

Of course, because of floating point issues the subtraction is a bad idea, and hence I corrected this in the last commit.